common/utils/src/main/resources/error/error-classes.json

```diff
@@ -298,6 +298,12 @@
     ],
     "sqlState" : "22018"
   },
+  "CANNOT_PARSE_STRING_AS_DATATYPE" : {
```

Comment:
Cannot you re-use the existing error class: `MALFORMED_RECORD_IN_PARSING.CANNOT_PARSE_STRING_AS_DATATYPE`?

Reply:
According to https://github.com/apache/spark/pull/45244/files#r1503550041, let me just turn this into an internal error.